daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -165,7 +166,7 @@ final class Engine {
     *  @param ledgerEffectiveTime time when the transaction is claimed to be submitted
     */
   def validate(
-      tx: Tx.Transaction,
+      tx: Tx.SubmittedTransaction,
```

Comment:
It's good to document that `validate` only works for submitted transaction, not for committed ones. 👏